mmedit/apis/inferencers/video_interpolation_inferencer.py

```diff
@@ -0,0 +1,245 @@
+            if self.extra_parameters['end_idx'] is not None else length
+
+        # calculate step args
+        step_size = \
+            self.model.step_frames * self.extra_parameters['batch_size']
+        lenth_per_step = self.model.required_frames + \
+            self.model.step_frames * (self.extra_parameters['batch_size'] - 1)
+        repeat_frame = self.model.required_frames - self.model.step_frames
+
+        prog_bar = ProgressBar(
+            math.ceil((self.extra_parameters['end_idx'] + step_size -
+                       lenth_per_step - self.extra_parameters['start_idx']) /
+                      step_size))
+        output_index = self.extra_parameters['start_idx']
+        for start_index in range(self.extra_parameters['start_idx'],
+                                 self.extra_parameters['end_idx'], step_size):
+            images = read_frames(
+                source,
+                start_index,
+                lenth_per_step,
+                from_video,
+                end_index=self.extra_parameters['end_idx'])
+
+            # data prepare
+            data = dict(img=images, inputs_path=None, key=inputs)
+            data = self.test_pipeline(data)['inputs'] / 255.0
+            data = collate([data])
+            # data.shape: [1, t, c, h, w]
+
+            # forward the model
+            data = self.model.split_frames(data)
+            input_tensors = data.clone().detach()
+            with torch.no_grad():
+                output = self.model(data.to(self.device), mode='tensor')
+                if len(output.shape) == 4:
+                    output = output.unsqueeze(1)
+                output_tensors = output.cpu()
+                if len(output_tensors.shape) == 4:
+                    output_tensors = output_tensors.unsqueeze(1)
+                result = self.model.merge_frames(input_tensors, output_tensors)
+            if not self.extra_parameters['start_idx'] == start_index:
+                result = result[repeat_frame:]
+            prog_bar.update()
+
+            # save frames
+            if to_video:
+                for frame in result:
+                    target.write(frame)
+            else:
+                filename_tmpl = self.extra_parameters['filename_tmpl']
+                for frame in result:
+                    save_path = osp.join(result_out_dir,
+                                         filename_tmpl.format(output_index))
+                    mmcv.imwrite(frame, save_path)
+                    output_index += 1
+
+            if start_index + lenth_per_step >= \
+               self.extra_parameters['end_idx']:
+                break
+
+        print()
+        print(f'Output dir: {result_out_dir}')
+        if to_video:
+            target.release()
+
+        return {}
+
+    def visualize(self,
+                  preds: PredType,
+                  result_out_dir: str = '') -> List[np.ndarray]:
+        """Visualize is not needed in this inferencer."""
+        pass
```

Comment:
Can we raise a warning or log here to tell users why we do not support `visualize` here?

Reply:
log added. resolved